nemo/collections/asr/data/audio_to_text_lhotse_target_language.py

```diff
@@ -0,0 +1,243 @@
+            )
+
+        index = self._GLOBAL_LANG_MAP[language_code]
+
+        return index
+
+    def lang_to_target_lang(
+        self, cut, num_languages: int, num_sample_per_mel_frame: int, num_mel_frame_per_asr_frame: int
+    ):
+        """
+        Create language target tensor for the sequence.
+        Similar to speaker_to_target but for languages.
+        """
+        # Calculate encoder output length
+        encoder_hidden_len = self.get_hidden_length_from_sample_length(
+            cut.num_samples, num_sample_per_mel_frame, num_mel_frame_per_asr_frame
+        )
+
+        # Initialize language target matrix
+        mask = np.zeros((num_languages, encoder_hidden_len))
+
+        # Get language index
+        lang_id = self._get_language_index(cut.supervisions[0].language)
+        # Set the corresponding language ID to 1 for all time steps
+        mask[lang_id, :] = 1
+
+        return mask
+
+    # this function is taken from nemo.collections.asr.parts.utils.asr_multispeaker_utils import get_hidden_length_from_sample_length
+    def get_hidden_length_from_sample_length(
+        self, num_samples: int, num_sample_per_mel_frame: int = 160, num_mel_frame_per_asr_frame: int = 8
+    ) -> int:
+        """
+        Calculate the hidden length from the given number of samples.
+
+        This function computes the number of frames required for a given number of audio samples,
+        considering the number of samples per mel frame and the number of mel frames per ASR frame.
+
+        Parameters:
+            num_samples (int): The total number of audio samples.
+            num_sample_per_mel_frame (int, optional): The number of samples per mel frame. Default is 160.
+            num_mel_frame_per_asr_frame (int, optional): The number of mel frames per ASR frame. Default is 8.
+
+        Returns:
+            hidden_length (int): The calculated hidden length in terms of the number of frames.
+        """
+        mel_frame_count = math.ceil((num_samples + 1) / num_sample_per_mel_frame)
+        hidden_length = math.ceil(mel_frame_count / num_mel_frame_per_asr_frame)
+        return int(hidden_length)
+
+    def __getitem__(self, cuts) -> Tuple[torch.Tensor, ...]:
+        audio, audio_lens, cuts = self.load_audio(cuts)
+        tokens = [torch.as_tensor(self.tokenizer(c.supervisions[0].text, c.supervisions[0].language)) for c in cuts]
+
+        # Create language targets
+        lang_targets = [
+            torch.transpose(
+                torch.as_tensor(
+                    self.lang_to_target_lang(
+                        c,
+                        self.num_languages,
+                        self.num_sample_per_mel_frame,
+                        self.num_mel_frame_per_asr_frame,
+                    ),
+                    dtype=torch.float32,
+                ),
+                0,
+                1,
+            )
+            for c in cuts
+        ]
+
+        # Create final tensors
```

Comment:
why to do this here and not part of training loop? That way you could avoid preparing these matrices and basically avoid this dataset all together.